src/Courses/TopicGrades/PrintEverything.tsx

```diff
@@ -6,52 +6,69 @@ import { ProblemObject } from '../CourseInterfaces';
 import url from 'url';
 import { getAllContentForVersion } from '../../APIInterfaces/BackendAPI/Requests/CourseRequests';
 import logger from '../../Utilities/Logger';
-import { useQuery } from '../../Hooks/UseQuery';
 import './PrintEverything.css';
 import { GetAllVersionAttachmentsResponse } from '../../APIInterfaces/BackendAPI/ResponseTypes/CourseResponseTypes';
 import PDFInlineRender from './PDFInlineRender';
+import { usePrintLoadingContext, PrintLoadingActions } from '../../Contexts/PrintLoadingContext';
+import OnLoadDispatchWrapper from './onLoadDispatchWrapper';
+import OnLoadProblemIframeWrapper from './OnLoadProblemIframeWrapper';
 
 interface PrintEverythingProps {
 }
 
 export const PrintEverything: React.FC<PrintEverythingProps> = () => {
-    let userId: number = 0;
-    let topicId: number = 0;
-    const qs = useQuery();
-    const workbookName = qs.get('workbookName');
+    const [gradeData, setGradeData] = useState<GetAllVersionAttachmentsResponse | null>(null);
+    const {dispatch} = usePrintLoadingContext();
     const params = useParams<{userId?: string, topicId?: string}>();
 
+    let userId: number = 0;
```

Comment:
I don't like defaulting to 0
Also there is an error that checks if they are nan and says they must be in the url, but if they are not it will let them through

Instead you could set them to null and down there check that they are both a number, you will accomplish the same thing without having a default value that could break the logic